lib/dry/types/compiler.rb

```diff
@@ -16,6 +16,11 @@ def visit(node)
         send(:"visit_#{ type }", body)
       end
 
+      def visit_constrained(node)
+        definition, rule, meta = node
+        Types::Constrained.new(visit(definition), rule: visit(rule)).meta(meta)
```

Comment:
Perfect. I will change it. I wasn't sure if that was going to work, because of the reduce(&:and) part. I thought each operation has to use a different reduce function